projects/storefrontapp-e2e-cypress/cypress/integration/product-search-e2e-spec.ts

```diff
@@ -0,0 +1,116 @@
+const PRODUCTS_PER_PAGE = 10;
+const resultsTitleSelector = 'header h1';
+const productItemSelector = 'cx-product-list cx-product-list-item';
+const firstProductItemSelector = `${productItemSelector}:first`;
+
+context('Product search', () => {
+  before(() => {
+    cy.visit('/');
+  });
+
+  describe('Search results', () => {
+    it('should be able to search and get results', () => {
+      cy.get('cx-searchbox input').type('camera{enter}');
+      cy.get(resultsTitleSelector).contains('results for camera');
+      cy.get(productItemSelector).should('have.length', PRODUCTS_PER_PAGE);
+      cy.get(firstProductItemSelector).within(() => {
+        cy.get('a.cx-product-name').should('be.visible');
```

Comment:
This can be changed to: `cy.get(firstProductItemSelector).find('a.cx-product-name').should('be.visible')` if we only check one assertion on it.